src/compressor/zlib/ZlibCompressor.h

```diff
@@ -17,14 +17,22 @@
 #ifndef CEPH_COMPRESSION_ZLIB_H
 #define CEPH_COMPRESSION_ZLIB_H
 
+#include "common/config.h"
 #include "compressor/Compressor.h"
 
 class ZlibCompressor : public Compressor {
   bool isal_enabled;
   CephContext *const cct;
 public:
   ZlibCompressor(CephContext *cct, bool isal)
-    : Compressor(COMP_ALG_ZLIB, "zlib"), isal_enabled(isal), cct(cct) {}
+    : Compressor(COMP_ALG_ZLIB, "zlib"), isal_enabled(isal), cct(cct) {
+#ifdef HAVE_QATZIP
+    if (cct->_conf->qat_compressor_enabled && qat_accel.init("zlib"))
```

Comment:
could you use the new style option? like `cct->_conf->get_val<bool>("qat_compressor_enabled")`?

Reply:
actually, this can be in the io path.  better to leave as is for now or use @rzarzynski's atomic<> wrapper class.

Reply:
Ok. this will leave as is and won't be changed.